Update scheduling-threads.md added an example code (#40739)
dotnet/docs: docs/standard/threading/scheduling-threads.md

## Example 

> Here is an example of the execution of 9 threads across all 5 priority levels from the [ThreadPriority](https://learn.microsoft.com/en-us/dotnet/api/system.threading.thread.priority) enumeration where the last 5 are on highest priority level. Also we have callback support from the previous article that in this context demonstrates that the order of thread initialization and prioritization may not always be reflected in subsequent code nor process executions start order. Meaning, we see here the parallel nature of code execution and demonstation of assigned processor time slices by the operating system for every thread. This highlights the influence and control of the environment in which threads are running. With that said, we certainly see that the highest priority threads are indeed receive priority in execution. 

Reviewer: Use an xref instead of an absolute link:

```suggestion
Here is an example of the execution of 9 threads across all 5 priority levels from the <xref:System.Threading.Thread.Priority?displayProperty=nameWithType> enumeration where the last 5 are on highest priority level. Also we have callback support from the previous article that in this context demonstrates that the order of thread initialization and prioritization may not always be reflected in subsequent code nor process executions start order. Meaning, we see here the parallel nature of code execution and demonstation of assigned processor time slices by the operating system for every thread. This highlights the influence and control of the environment in which threads are running. With that said, we certainly see that the highest priority threads are indeed receive priority in execution. 
```

The following code will output arbitrary results on each execution. But common sequence patters of priorities being ran could be observed after running multiple times and analyzing the outputs.

> ```c#

Reviewer: For samples, we've been moving toward including samples as code files using the `:::` code extension. That enables our CI system to validate the code. Can you move the sample code into a project and include the code here?

Reviewer: Hi @victoravt 

I apologize for the late reply. I missed this, and was out on vacation.

Add new snippets as a project in a `snippets` folder in the current folder. Relative to the root of the repo, it would be `docs/standard/threading/snippets/scheduling-threads`

Thanks.

Author: Hi @BillWagner 
Thanks for the replay. Please review the changes (moved code snippet to docs/standard/threading/snippets/scheduling-threads/SchedulingThreads.cs with .csproj).
Thank you!

Author: Hey @BillWagner.
Thanks for the comments.
I fixed the white spaces and rephrased couple of sentences.

            Thread.Sleep(1000);
            Console.WriteLine($"Finished process in {Thread.CurrentThread.Priority} priority. \t\t ThreadId: {Thread.CurrentThread.ManagedThreadId}.");

            if (callback != null)
            {
                callback(Thread.CurrentThread.Priority);
            }
        }
    }

    public void RunMultipleThreadsOnDifferentPriorities()
    {
        Thread t0 = new Thread(() => { new ThreadWithCallback(Callback).Process(); });
        Thread t1 = new Thread(() => { new ThreadWithCallback(Callback).Process(); });
        Thread t2 = new Thread(() => { new ThreadWithCallback(Callback).Process(); });
        Thread t3 = new Thread(() => { new ThreadWithCallback(Callback).Process(); });
        Thread t4 = new Thread(() => { new ThreadWithCallback(Callback).Process(); });
        Thread t5 = new Thread(() => { new ThreadWithCallback(Callback).Process(); });
        Thread t6 = new Thread(() => { new ThreadWithCallback(Callback).Process(); });
        Thread t7 = new Thread(() => { new ThreadWithCallback(Callback).Process(); });

>         Thread t8 = new Thread(() => { new ThreadWithCallback(Callback).Process(); });

Reviewer: There's a log of opportunity to refactor this so it's got less repeated code. Maybe store the threads in an array and loop through the creation, setting priority, and start code?

> ## Example 

Reviewer: Remove trailing space, add blank line:

```suggestion
## Example

```

Author: Done

> Here is an example of the execution of 9 threads across all 5 priority levels from the <xref:System.Threading.Thread.Priority?displayProperty=nameWithType> enumeration where the last 5 are on highest priority level. Also we have callback support from the previous article that in this context demonstrates that the order of thread initialization and prioritization may not always be reflected in subsequent code nor process executions start order. Meaning, we see here the parallel nature of code execution and demonstration of assigned processor time slices by the operating system for every thread. This highlights the influence and control of the environment in which threads are running. With that said, we certainly see that the highest priority threads are indeed receive priority in execution. 

Reviewer: remove extra space:

```suggestion
Here is an example of the execution of 9 threads across all 5 priority levels from the <xref:System.Threading.Thread.Priority?displayProperty=nameWithType> enumeration where the last 5 are on highest priority level. Also we have callback support from the previous article that in this context demonstrates that the order of thread initialization and prioritization may not always be reflected in subsequent code nor process executions start order. Meaning, we see here the parallel nature of code execution and demonstration of assigned processor time slices by the operating system for every thread. This highlights the influence and control of the environment in which threads are running. With that said, we certainly see that the highest priority threads are indeed receive priority in execution.
```

Author: Done
